rero_ils/modules/patrons/api.py

```diff
@@ -122,6 +122,39 @@ class Patron(IlsRecord):
         UserDataExtension()
     ]
 
+    def extended_validation(self, **kwargs):
+        """Returns reasons for validation failures, otherwise True.
+
+        Ensures that barcode field is unique in the organisation.
+
+        :return: Error message if
+            - barcode already exists in organisation
+        """
+        org_pid = self.organisation_pid
+        if patron_barcodes := self.get('patron', {}).get('barcode', []):
+            filters = Q('terms', patron__barcode=patron_barcodes)
+            filters &= Q('term', organisation__pid=org_pid)
+            results = PatronsSearch()\
+                .filter('bool', must=[filters])\
+                .source(['pid', 'patron.barcode'])\
+                .scan()
+            taken_barcodes = []
+            for hit in results:
+                if hit.pid != self.get('pid'):
+                    taken_barcodes.extend(
+                        barcode
+                        for barcode in hit.patron.barcode
+                        if barcode in patron_barcodes
+                    )
```

Comment:
To be tested 

```python
taken_barcodes = [
   barcode
   for barcode in patron_barcodes
   if (patron := Patron.get_patron_by_barcode(barcode, org_pid)) and patron.pid != self.get('pid')
]
```

additionally, seems strange too to have possibility to have `None` as barcode parameter value for `get_patron_by_barcode` (don't "blame" who wrote this :-) I just reorganized the class - I didn't write this method ^^)

Reply:
@zannkukai This might work but it creates multiple API calls if the patron has multiple barcodes instead of one ES query.

Reply:
you're right ; but in my opinion it's better to delegate the search job to a specific method instead of write the query here. Maybe the best solution is to adapt `get_patron_by_barcode()` method to accept one or multiple barcodes parameters... See with other devs opinion (@rerowep, @lauren-d, @jma, @Garfield-fr)